src/scmap.rs

```diff
@@ -0,0 +1,76 @@
+#![allow(clippy::missing_errors_doc)]
+
+use crate::{Error, ScMap, ScMapEntry, ScVal, Validate};
+extern crate alloc;
+use alloc::vec::Vec;
+
+impl ScMap {
+    pub fn sorted_from_entries<I, E>(entries: I) -> Result<ScMap, Error>
+    where
+        E: TryInto<ScMapEntry>,
+        I: Iterator<Item = E>,
+    {
+        let mut v = entries
+            .map(TryInto::try_into)
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|_| Error::Invalid)?;
+        // TODO: Add tests that prove order consistency of ScVal with RawVal. https://github.com/stellar/rs-stellar-xdr/issues/117
+        v.sort_by(|a, b| a.key.cmp(&b.key));
+        let m = ScMap(v.try_into()?);
+        // `validate` will further check that there are no duplicates.
+        m.validate()?;
+        Ok(m)
+    }
+
+    pub fn sorted_from_pairs<K, V, I>(pairs: I) -> Result<ScMap, Error>
+    where
+        K: TryInto<ScVal>,
+        V: TryInto<ScVal>,
+        I: Iterator<Item = (K, V)>,
+    {
+        Self::sorted_from_entries(pairs)
+    }
```

Comment:
This fn is now technically redundant because the same values can be passed directly to the entries fn. But this fn has a clearer signature that is easier to understand, so I think it is still useful so I kept it.